sdk/azidentity/azure_cli_credential.go

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+type AzureCLITokenProvider interface {
+	GetCLIToken(ctx context.Context, resource string) ([]byte, error)
+}
+
+type azureCLITokenProvider struct{}
+
+// AzureCLICredentialOptions contains options used to configure the AzureCLICredential
+type AzureCLICredentialOptions struct {
+	TokenProvider AzureCLITokenProvider
+}
+
+// AzureCLICredential enables authentication to Azure Active Directory using the Azure CLI command "az account get-access-token".
+type AzureCLICredential struct {
+	tokenProvider AzureCLITokenProvider
+}
+
+// TODO: do we want an options bag for this credential that includes a developer specified Azure CLI path or only specify with env var?
```

Comment:
Is this comment still relevant?  Looks like options are now in the constructor's signature.

Reply:
I wanted to get a general opinion on this. I think it's important so that we can add the alternate TokenProvider.

Reply:
Removed the comment, since we'll have the alternate token provider in the options struct